python_modules/jupedsim/jupedsim/sqlite_serialization.py

```diff
@@ -174,12 +191,49 @@ def write_iteration_state(self, simulation: Simulation) -> None:
                     ("ymax", str(max(ymax, float(old_ymax)))),
                 ],
             )
+            if not self._buffer_in_memory:
+                cur.execute("COMMIT")
+            else:
+                # Trigger flush if buffer full
+                self._buffered_frame_count += 1
+                if self._buffered_frame_count >= self._max_buffered_frames:
+                    self.flush()
+                return
+        except sqlite3.Error as e:
+            cur.execute("ROLLBACK")
+            raise TrajectoryWriter.Exception(f"Error writing to database: {e}")
+
 
+    def flush(self) -> None:
+        """Flush any buffered frames to disk in a single transaction.
+
+        Safe to call multiple times; if the buffer is empty, this won't execute.
+        """
+        if not self._buffer_in_memory:
+            return
+        if self._buffered_frame_count == 0:
+            return
+        cur = self._con.cursor()
+        try:
```

Comment:
I don't think we need this, since with no hand written buffering this should(TM) be only a `execute("COMMIT")`